### PR TITLE
feat: added clickup connection

### PR DESCRIPTION
**Pull request by IbrahimMohammed47**
In this pull request I tried to add integration for Clickup.. I haven't added frontend code for this integration, I just added the connection code in server as stated in [CONTRIBUTING.md](https://github.com/psychic-api/psychic/blob/main/CONTRIBUTING.md). 

The main "Document" here is a Clickup Task, so I help users import their tasks (from any path) into psycho. I haven't found any Clickup API endpoints to query "document" objects from a clickup space. So I handled tasks only.

I tested most of my code but in isolated way, I'm not sure If it would work with rest of application, but I will try to test that as soon as possible.

This is my first pull request to an open source project
Thank you

### CODE REVIEW

File: server/connectors/connector_utils.py [hunk 1]
```diff
@@ -58,4 +61,6 @@ def get_connector_for_id(connector_id: ConnectorId, config: AppConfig) -> Option
         return ReadmeConnector(config)
     elif connector_id == ConnectorId.salesforce:
         return SalesforceConnector(config)
+    elif connector_id == ConnectorId.clickup:
```
Comment (kapilsinha) [inline server/connectors/connector_utils.py:64]: Probably cleaner (and negligibly more performant) to refactor this into a dictionary of ConnectorId -> ConnectorFactoryFunction. I think Python 3.10 has a match + case, but apparently it's not quite a switch

Comment (IbrahimMohammed47) [reply server/connectors/connector_utils.py:64]: True, However I wanted my pull request to add just one feature, which is Clickup support. I think there is room for a lot of refactoring here, even in my code itself.

**Comment by Ayan-Bandyopadhyay** [comment]
Thanks for the PR! I'll take a look this weekend

**Comment by Ayan-Bandyopadhyay** [comment]
@IbrahimMohammed47  Given that clickup tasks are like tickets that can have comments, responses, etc. it might make more sense to have this be a `ConversationConnector` that returns a list of messages w/ response, sender, timestamp, etc. See the slack connector for an example.

**Comment by Ayan-Bandyopadhyay** [comment]
Actually ignore my previous comment, we're going to be deprecating the conversations connector and merging it into the document connector

**Review by Ayan-Bandyopadhyay — APPROVED** [review]

**Comment by jasonwcfan** [comment]
Hey @IbrahimMohammed47 are you on twitter? Would love to give you a shoutout for the contribution!

**Comment by IbrahimMohammed47** [comment]
Hi @jasonwcfan , I am not on twitter, I just have [Linkedin](https://www.linkedin.com/in/ibrahimmohamed47)  now.
Thanks for consideration

**Pull request merged**